pg_search/src/postgres/utils.rs

```diff
@@ -235,3 +237,32 @@ pub fn convert_pg_date_string(typeoid: PgOid, date_string: &str) -> tantivy::Dat
         _ => panic!("Unsupported typeoid: {typeoid:?}"),
     }
 }
+
+pub trait CoercibleTo {
+    fn is_coercible_to(&self, builtin: PgBuiltInOids) -> bool;
```

Comment:
I'm not 100% satisfied with the name. I considered `coercible` as well as `compatible`. The only thing that bothers me about `coercible` is that for built-in types, we're doing an exact match even though technically there are built-in types that are binary coercible to other built-in types (e.g. `text` and `varchar` in either direction). Let me know if you think `compatible` or something else would be a better term to use here for our intent, which is to check if an OID can be represented by some other specified built-in OID.